source/admin_manual/migrating_mailboxes.rst

```diff
@@ -0,0 +1,200 @@
+ * Migration can cause unexpectedly high loads on the source system, beware. There is no way to throttle the synchronization.
+
+.. _migrating_mailboxes_imapc:
+
+Migrating mailboxes over IMAP
+=============================
+
+When migrating mails over IMAP, you need to have valid credentials to the source system. You can either user master password, master user or individual user authentication. 
+This depends entirely on the source system. See :ref:`authentication-master_users` for more details.
+
+Preparations
+------------
+
+Configure authentication on the source system to match your preference. Ensure both source and target system agree on usernames.
+
+Configure IMAP client on the target system.
+
+Common settings::
+
+  # this is sometimes needed with some broken servers
+  #dsync_features = empty-header-workaround
+  
+  # if your source host supports these, set them
+  #imapc_features = rfc822.size fetch-headers
+
+  imapc_host = remotehost
+
+  ## if you are using TLS
+  imapc_ssl = imaps
+  imapc_port = 993
+
+  # if you are using StartTLS
+  imapc_ssl = starttls
+  imapc_port = 143
+
+  ## these default to system 
+  ssl_client_ca_dir = /etc/ssl/certs
+  # or
+  ssl_client_ca_file = /etc/ssl/ca-certificates.pem
+  
+Master password::
+
+  imapc_user = %u
+  imapc_password = supersecret
+
+Master user::
+
+  imapc_user = %u
+  imapc_master_user = master-user
+  imapc_password = master-password
+
+Individual password::
+
+  imapc_user = %u
+
+  doveadm -c /etc/dovecot/dovecot-migration.conf -o imapc_password=password sync ...
+
+You can verify that the settings are done correctly with::
+
+  doveadm -c /etc/dovecot/dovecot-migration.conf -o mail_location=imapc: mailbox list -u user
+
+POP3 notes
+----------
+
+If you need to retain POP3 support on your new system, you should use :ref:`plugin-pop3-migration`.
+
+::
+
+  pop3c_host = hostname
+
+  # if you are using TLS
+  pop3c_ssl = pop3s
+  pop3c_port = 995
+
+  # if you are using StartTLS
+  pop3c_ssl = starttls
+  pop3c_port = 110
+
+  # see imapc migration for
+
+  namespace {
+    prefix = POP3-MIGRATION-NS/
+    location = pop3c:
+    list = no
+    hidden = yes
+  }
+
+  protocol doveadm {
+    mail_plugins = $mail_plugins pop3_migration
+  }
+
+  plugin {
+    pop3_migration_mailbox = POP3-MIGRATION-NS/INBOX
+  }
+
+You can alternatively configure compatible UIDL format in Dovecot. See :ref:`setting-pop3_uidl_format`.
+
+Executing migration
+-------------------
+
+To migrate users, use::
+
+  doveadm -c /etc/dovecot/dovecot-migration.conf sync -1Ru username imapc:
+
+If you are experiencing problems, run::
+
+  doveadm -D -c /etc/dovecot/dovecot-migration.conf sync -1Ru username imapc:
+
+This will enable debug logging.
+
+Note that Public and Shared namespaces are synchronized automatically (see caveats).
+
+Caveats
+-------
+
```

Comment:
There's quite a lot of useful stuff in https://wiki.dovecot.org/Migration and its sub-pages that is missing here. We don't need to copy everything though (especially not the obsolete ways of doing things).

Reply:
Sure, can you tell me which bits are useful?

Reply:
To me it seemed most of them were only useful if you are replacing, say, UW-IMAP with Dovecot without migration?

Reply:
IMAP migration
 * The list "When migrating mails from another IMAP server, you should make sure that these are preserved: "
 
POP3 migration
  *  If a client already saw changed UIDLs and decided to start re-downloading mails, it's unlikely there is anything you can do to stop it. Even going back to your old server is unlikely to help at that point.
  *  Some (many?) POP3 clients also require that the message ordering is preserved.
  *  Some clients re-download all mails if you change the hostname in the client configuration. Be aware of this when testing. 

The whole of https://wiki.dovecot.org/Migration/Gmail

From https://wiki.dovecot.org/Migration/Dsync 
 * Generic configuration
 * Problems (although "currently preserved only when destination is Maildir, sdbox or mdbox" -> "not preserved with mbox format")